include/dev/usb.h

```diff
@@ -145,4 +149,24 @@ typedef struct usb_endpoint_descriptor {
 #define UE_BULK 0x02
 #define UE_INTERRUPT 0x03
 
+/*
+ * Implementation specific constructs.
+ */
+
+typedef enum {
+  TF_CONTROL = 1,   /* control transfer */
+  TF_BULK = 2,      /* bulk transfer */
+  TF_INTERRUPT = 4, /* interrupt transfer */
+  TF_STALLED = 8,   /* STALL condition encountered */
+  TF_ERROR = 16,    /* errors other than STALL */
+} usb_transfer_flags_t;
```

Comment:
Aren't _control_, _bulk_ and _interrupt_ transfers mutually exclusive? Personally I would've made a comment for an enum which contains OR-able flags with a mutually exclusive subset but I'm leaving that decision up to you.

Reply:
The transfer flags are gone. I contracted it to the error flags and renamed to `usb_error_flags_t`.